pennylane/ops/op_math/adjoint_class.py

```diff
@@ -241,12 +239,7 @@ def adjoint(self):
         return self.base.queue()
 
     def simplify(self):
-        if isinstance(self.base, qml.Identity):
-            return self.base
-        if isinstance(self.base, Adjoint):  # Adj(Adj(A)) = A
-            return self.base.base.simplify()
-        if isinstance(self.base, Sum):  # Adj(A + B) = Adj(A) + Adj(B)
-            return Sum(*(Adjoint(summand) for summand in self.base.summands)).simplify()
-        if isinstance(self.base, Prod):  # Adj(AB) = Adj(B) @ Adj(A)
-            return Prod(*(Adjoint(factor) for factor in self.base.factors[::-1])).simplify()
-        return Adjoint(base=self.base.simplify())
+        try:
+            return self.base.adjoint().simplify()
+        except AdjointUndefinedError:
+            return Adjoint(base=self.base.simplify())
```

Comment:
How does this behave for the case Adj(Adj(A)) where A doesn't have an adjoint method defined?

Reply:
But `Adj(A)` does have an adjoint defined, `A`.

Reply:
Perfect!